File: app/src/main/java/com/example/monappliimmo/BatimentDetailActivity.java
package com.example.monappliimmo;

import android.os.Bundle;
import android.view.MenuItem;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class BatimentDetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_batiment_detail);

        // 1. Configuration de la Toolbar avec flèche de retour
        Toolbar toolbar = findViewById(R.id.toolbar_batiment);
        setSupportActionBar(toolbar);

        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setDisplayShowHomeEnabled(true);
            getSupportActionBar().setTitle("Détails de l'immeuble");
        }

        // 2. Récupération du bâtiment
        Batiment batiment = (Batiment) getIntent().getSerializableExtra("BATIMENT_OBJ");

        if (batiment != null) {
            TextView tvAdresse = findViewById(R.id.tvDetailAdresse);
            TextView tvVille = findViewById(R.id.tvDetailVille);

            tvAdresse.setText(batiment.getAdresse());
            tvVille.setText(batiment.getVille());

            // 3. Configuration de la liste des appartements
            RecyclerView rv = findViewById(R.id.rvAppartsDuBatiment);
            rv.setLayoutManager(new LinearLayoutManager(this));

            if (batiment.getAppartements() != null) {
                // IMPORTANT : On passe la liste d'apparts ET l'objet batiment lui-même
                AppartementAdapter adapter = new AppartementAdapter(batiment.getAppartements(), batiment);
                rv.setAdapter(adapter);
            }
        }
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/example/monappliimmo/BatimentDetailActivity.kt
package com.example.monappliimmo

import android.os.Bundle
import android.view.MenuItem
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class BatimentDetailActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_batiment_detail)

        // 1. Configuration de la Toolbar avec flèche de retour
        val toolbar = findViewById<Toolbar?>(R.id.toolbar_batiment)
        setSupportActionBar(toolbar)

        if (supportActionBar != null) {
            supportActionBar!!.setDisplayHomeAsUpEnabled(true)
            supportActionBar!!.setDisplayShowHomeEnabled(true)
            supportActionBar!!.title = "Détails de l'immeuble"
        }

        // 2. Récupération du bâtiment
        val batiment = intent.getSerializableExtra("BATIMENT_OBJ") as Batiment?

        if (batiment != null) {
            val tvAdresse = findViewById<TextView>(R.id.tvDetailAdresse)
            val tvVille = findViewById<TextView>(R.id.tvDetailVille)

            tvAdresse.text = batiment.adresse
            tvVille.text = batiment.ville

            // 3. Configuration de la liste des appartements
            val rv = findViewById<RecyclerView>(R.id.rvAppartsDuBatiment)
            rv.layoutManager = LinearLayoutManager(this)

            if (batiment.appartements != null) {
                // CORRECTION : On utilise .filterNotNull() pour transformer
                // MutableList<Appartement?>? en List<Appartement>
                val listeFiltree = batiment.appartements!!.filterNotNull()

                val adapter = AppartementAdapter(listeFiltree, batiment)
                rv.adapter = adapter
            }
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}